### PR TITLE
Add unit tests for CognitoUtils class

### LINKED ISSUE #561: write unit tests for CognitoUtils.js
has an integration test, since sign-in is exercised, but the individual functions should be given unit tests for their basic success and failure modes.

spawned by #528

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #561

### CODE REVIEW

File: __tests__/CognitoUtils.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  })
+
+  describe('authenticateUser()', () => {
+    const user = CognitoUtils.cognitoUser(username)
+
+    afterEach(() => {
+      jest.restoreAllMocks()
+    })
+
+    it('returns a promise', () => {
```
Comment (ndushay) [inline __tests__/CognitoUtils.test.js]: 👍

File: __tests__/CognitoUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      const getSessionSpy = jest.spyOn(user, 'getSession').mockImplementation(callback => {
+        return callback(mockError, mockSession)
+      })
+      return CognitoUtils.getIdTokenString(user)
+        .then(result => {
+          expect(getSessionSpy).toHaveBeenCalledTimes(1)
+          expect(result).toEqual(token)
+        })
+        .catch(error => { throw error })
+    })
```
Comment (jmartin-sul) [inline __tests__/CognitoUtils.test.js]: test for the failure case might be nice, but wouldn't hold up merging over it.

File: __tests__/CognitoUtils.test.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      expect(pool.clientId).toEqual(Config.awsClientID)
+    })
+  })
+
+  describe('cognitoUser()', () => {
+    it('returns an instance of CognitoUser with username and user pool', () => {
+      const user = CognitoUtils.cognitoUser(username)
+      expect(user).toBeInstanceOf(CognitoUser)
+      expect(user.username).toEqual(username)
+      expect(user.pool).toBeInstanceOf(CognitoUserPool)
```
Comment (jmartin-sul) [inline __tests__/CognitoUtils.test.js]: i would be slightly inclined to check that the pool has the right details... but that might be overkill?

File: __tests__/CognitoUtils.test.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    })
+
+    it('returns a promise', () => {
+      const authResult = CognitoUtils.authenticateUser(user, password)
+      expect(authResult).toBeInstanceOf(Promise)
+    })
+
+    it('resolves on successful authentication', () => {
+      const mockSession = {}
+      const authUserSpy = jest.spyOn(user, 'authenticateUser').mockImplementation((_details, callback) => {
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/CognitoUtils.test.js]: Good catch. Yeah, maybe `callbacksObject` to surface that the param is a legit object.

File: __tests__/CognitoUtils.test.js (hunk 2 above)
Comment (mjgiarlo) [reply __tests__/CognitoUtils.test.js]: Would be good to add. What would/should that function do when there's a failure? I don't believe it catches.

Comment (jmartin-sul) [reply __tests__/CognitoUtils.test.js]: i think just bubble up the original error, though i hadn't thought about it much. we'll see if a test that effect ends up exposing weird behavior or something.

**Pull request merged**